src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -823,7 +823,19 @@ private function syncNode($uuid, $path, $type, $isNewNode, $props = array(), $pr
             $this->syncBinaryData($nodeId, $propsData['binaryData']);
         }
 
-        if (!empty($propsData['references'])) {
+        $needSyncReferences = !empty($propsData['references']);
+        if (!$needSyncReferences) {
+            foreach ($props as $property) {
+                if (in_array($property->getType(), array(PropertyType::REFERENCE, PropertyType::WEAKREFERENCE))) {
+                    // there are no references from this node anymore, which means there is a multivalue
+                    // reference property with an empty array. need to sync to remove previous references
```

Comment:
i think this was a regression edge case when supporting empty arrays in multivalue properties. previously we considered the property to be deleted (and deleting a reference property does clean up the references)

but actually, there is some more problems in here: [setting a property to empty array changes its type to string](https://github.com/jackalope/jackalope/pull/295). but even that fix is not enough. we might change a property type actively so its no longer a reference. this is a weird edge case of course, but probably we should just unconditionally sync references if any property changed, instead of trying to be clever.